mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/Task.java

```diff
@@ -157,4 +174,16 @@ public Observable<VirtualMachineTaskStatus> getVMStatus() {
     public WorkerId getWorkerId() {
         return executeStageRequest.getWorkerId();
     }
+
+    private void closeUserCodeClassLoader() {
+        if (userCodeClassLoader != null) {
+            if (userCodeClassLoader.asClassLoader() instanceof Closeable) {
+                try {
+                    ((Closeable) userCodeClassLoader.asClassLoader()).close();
+                } catch (IOException ex) {
+                    log.error("Failed to close user class loader successfully", ex);
+                }
+            }
+        }
+    }
```

Comment:
hmm, the task is not opening the class loader - so why is it given the responsibility of closing the class loader? Should this be part of the actor that's actually opening the class loader?